components/basecomponents/Story/StoriesList.tsx

```diff
@@ -24,73 +24,69 @@ const StoriesList = ({ stories }: StoriesListProps) => {
+  const RenderStoryPanel = ({ story }: { story: Story }) => {
+    // Check if the current image's source is "/kids/dvanact-mesicku.jpg", because it takes a lot of time to load
+    const isPriorityImage = story.slug === 'dvanact-mesicku';
 
-  const renderStoryPanel = (story: Story) => (
-    <div className="h-42 m-auto my-8 flex bg-slate-50 rounded-2xl sm:w-3/5 xl:w-2/5 xxl:w-1/3" key={story.slug}>
-      <div className="relative w-[105px] lg:h-[200px] lg:w-[300px]">
-        <Image src={`/kids/${story.slug}.jpg`} fill className="rounded-l-2xl" alt={story.title[currentLanguage]} />
-      </div>
-      <div className="flex items-center xl:ml-6 w-full">
-        <div className="w-3/5">
-          <p className="my-4 mx-4 md:mx-8 text-sm lg:text-xl ">{story.title[currentLanguage]}</p>
-          <div className="flex items-center mb-2">
-            <svg
-              xmlns="http://www.w3.org/2000/svg"
-              className="h-5 w-5 mx-2 md:ml-8 text-sm lg:text-xl"
-              fill="none"
-              viewBox="0 0 24 24"
-              stroke="currentColor"
-              strokeWidth={2}
-            >
-              <path strokeLinecap="round" strokeLinejoin="round" d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
-            </svg>
-            <span className="text-sm lg:text-xl">{story.duration}</span>
-          </div>
+    return (
+      <div className="max-w-[342px] sm:max-w-none h-[376px] mb-[32px] sm:w-[400px] sm:h-[400px] sm:mb-[48px] group">
+        <div className="rounded-t-2xl relative h-[300px]">
+          <Image
+            src={`/kids/${story.slug}.jpg`}
+            width={400}
+            height={300}
+            className="w-full h-full object-cover rounded-t-2xl"
+            alt={story.title[currentLanguage]}
+            priority={isPriorityImage}
+          />
+          <Link href={`/kids/stories/${story.slug}`} className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2">
+            <PlayIcon className="w-full cursor-pointer w-[60px] h-[60px] sm:w-[74.57px] sm:h-[74.57px] group-hover:visible xl:invisible inline-block active:scale-125 active:fill-primary-yellow active:stroke-primary-yellow transition duration-100 fill-primary-blue stroke-primary-blue" />
+          </Link>
         </div>
-
-        <Link
-          href={`/kids/stories/${story.slug}`}
-          className="w-2/5 bg-primary-blue rounded-2xl text-white text-[10px] sm:text-[12px] lg:text-base p-2 w-3/5 text-center mr-4"
-        >
-          {t('kids_page.playStory')}
-        </Link>
+        <p className="bg-[#FFFFFF] text-primary-blue text-center text-2xl rounded-b-2xl flex justify-center text-center h-[76px] sm:h-[100px] items-center">
+          {story.title[currentLanguage]}
+        </p>
       </div>
-    </div>
-  );
+    );
+  };
 
-  const renderStoriesSection = (language: string, stories: Story[], titleKey: string) => {
+  const RenderStoriesSection = ({
```

Comment:
Asi by stálo za to zjistit zda je lepší použít Functional Component nebo obyčejnou funkci.

Reply:
no tak mě přijde obyčejná funkce naprosto nestandartní, proto jsem to přepsal na FC, i mi to Martin Starosta schválil

Reply:
Před pár lety se upřednostňovaly před FC. Teď už to má nejspíš react optimalizováno tak že je lépe použít FC.